src/crimson/osd/pg.cc

```diff
@@ -753,12 +753,22 @@ PG::do_osd_ops(
       Ref<PG>{this}, obc, op_info, *m),
     m->ops,
     op_info,
-    [this, m, obc, may_write = op_info.may_write(), rvec = op_info.allows_returnvec()] {
+    [this, m, obc, may_write = op_info.may_write(),
+     may_read = op_info.may_read(), rvec = op_info.allows_returnvec()] {
       // TODO: should stop at the first op which returns a negative retval,
       //       cmpext uses it for returning the index of first unmatched byte
       int result = m->ops.empty() ? 0 : m->ops.back().rval.code;
```

Comment:
Restructure this as if -- else if -- else

Reply:
done.